contracts/liquidity_hub/fee_distributor/src/tests/epoch.rs

```diff
@@ -30,3 +32,40 @@ fn test_expiring_epoch() {
         .add_epochs_to_state(epochs.clone())
         .assert_expiring_epoch(Some(&epochs[1]));
 }
+
+#[test]
+fn test_create_genesis_epoch() {
+    let mut robot = TestingRobot::new(mock_dependencies(), mock_env());
+    let grace_period = Uint64::new(2);
+    let distribution_asset = AssetInfo::NativeToken {
+        denom: "uwhale".to_string(),
+    };
+    let epoch_config = EpochConfig {
+        duration: Uint64::new(86_400_000_000_000u64), // a day
```

Comment:
Could be im missing something but why nanos everywhere, is it related to the implementation?

Reply:
Yeah we are using nanos cuz timestamp is serialized to nanos on the frontend, so initially we had seconds but then it was mixed cuz some configurations where in seconds while the timestamps where returning nanoseconds. To be consistent we just made everything nanoseconds.